app/audit.go

```diff
@@ -14,3 +38,112 @@ func (a *App) GetAudits(userId string, limit int) (model.Audits, *model.AppError
 func (a *App) GetAuditsPage(userId string, page int, perPage int) (model.Audits, *model.AppError) {
 	return a.Srv().Store.Audit().Get(userId, page*perPage, perPage)
 }
+
+func (s *Server) configureAudit(adt *audit.Audit) {
+	// For now just send audit records to database. When implemented, use config store
+	// to configure where audit records are written, and which filter is used.
+	filter := adt.MakeFilter(RestLevel, RestContentLevel, RestPermsLevel, CLILevel)
+	target := NewAuditStoreTarget(filter, s.Store.Audit(), audit.DefMaxQueueSize)
+	adt.AddTarget(target)
+
+	adt.OnQueueFull = s.onAuditTargetQueueFull
+	adt.OnError = s.onAuditError
+}
+
+func (s *Server) onAuditTargetQueueFull(qname string, maxQSize int) {
+	mlog.Warn("Audit Queue Full", mlog.String("qname", qname), mlog.Int("maxQSize", maxQSize))
+}
+
+func (s *Server) onAuditError(err error) {
+	mlog.Error("Audit Error", mlog.String("err", err.Error()))
```

Comment:
We can do mlog.Err(err) instead of mlog.String("err"...)

Reply:
The error in this case could really be anything as it likely is coming from one of the custom targets. As such I would like the log output to include the "Audit Error" for searching and something to alert on.  I guess I could wrap the error, but this seems simpler. If you have a strong opinion on it I will wrap it to get the "Audit Error" key word.

Reply:
Sorry, you misunderstood. I meant: `mlog.Error("Audit Error", mlog.Err(err))`.

Reply:
Ah right, I see.  Will fix.

Reply:
Fixed via https://github.com/mattermost/mattermost-server/pull/13967/commits/71165f9a26f3bc7f166551c519118312b91dafc2    Thanks!